libcaf_core/src/detail/rfc3629.cpp

```diff
@@ -49,57 +54,75 @@ bool validate_rfc3629(const std::byte* first, const std::byte* last) {
     if (head<3>(x) == 0b1100'0000_b) {
       // No non-shortest form.
       if (head<7>(x) == 0b1100'0000_b)
-        return false;
-      if (first == last || !is_continuation_byte(*first++))
-        return false;
+        return std::pair(checkpoint, false);
```

Comment:
1) We prefer `{}` over `()` for initialization: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es23-prefer-the--initializer-syntax.
2) We don't need to repeat the type name of the return type here:

```suggestion
        return {checkpoint, false};
```

Just tagging it here, but it applies to multiple sites.